netbox/utilities/forms/utils.py

```diff
@@ -48,53 +48,41 @@ def parse_alphanumeric_range(string):
     Expand an alphanumeric range (continuous or not) into a list.
     'a-d,f' => [a, b, c, d, f]
     '0-3,a-d' => [0, 1, 2, 3, a, b, c, d]
+    '9-11' => [9, 10, 11]
     """
     values = []
     for dash_range in string.split(','):
-        try:
-            begin, end = dash_range.split('-')
-            vals = begin + end
-            # Break out of loop if there's an invalid pattern to return an error
-            if (not (vals.isdigit() or vals.isalpha())) or (vals.isalpha() and not (vals.isupper() or vals.islower())):
-                return []
-        except ValueError:
-            begin, end = dash_range, dash_range
-        if begin.isdigit() and end.isdigit():
-            if int(begin) >= int(end):
-                raise forms.ValidationError(f'Range "{dash_range}" is invalid.')
-
-            for n in list(range(int(begin), int(end) + 1)):
-                values.append(n)
+        if re.fullmatch(ALPHABETIC_RANGE_PATTERN, dash_range):
+            begin, end = map(ord, dash_range.split('-'))
+            if begin > end:
+                raise forms.ValidationError(f'Range "{dash_range}" is invalid, because {begin} comes after {end}')
+            values.extend(map(chr, range(begin, end + 1)))
+        elif re.fullmatch(NUMERIC_RANGE_PATTERN, dash_range):
+            begin, end = map(int, dash_range.split('-'))
+            if begin > end:
+                raise forms.ValidationError(f'Range "{dash_range}" is invalid, because {begin} comes after {end}')
+            values.extend(map(str, range(begin, end + 1)))
+        elif re.fullmatch(ALPHANUMERIC_SINGLETON_PATTERN, dash_range):
+            values.append(dash_range)
         else:
-            # Value-based
-            if begin == end:
-                values.append(begin)
-            # Range-based
-            else:
-                # Not a valid range (more than a single character)
-                if not len(begin) == len(end) == 1:
-                    raise forms.ValidationError(f'Range "{dash_range}" is invalid.')
-
-                if ord(begin) >= ord(end):
-                    raise forms.ValidationError(f'Range "{dash_range}" is invalid.')
-
-                for n in list(range(ord(begin), ord(end) + 1)):
-                    values.append(chr(n))
+            raise forms.ValidationError(f'Range "{dash_range}" is invalid, must be a range of numbers (e.g. 7-11), a range of letters (e.g. f-h or F-H), or a single alphanumeric string (e.g. abc1234)')
+
     return values
 
 
 def expand_alphanumeric_pattern(string):
     """
     Expand an alphabetic pattern into a list of strings.
     """
-    lead, pattern, remnant = re.split(ALPHANUMERIC_EXPANSION_PATTERN, string, maxsplit=1)
+    try:
+        lead, pattern, remnant = re.split(ALPHANUMERIC_EXPANSION_PATTERN, string, maxsplit=1)
+    except ValueError:
+        yield string
+        return
```

Comment:
If I understand you correctly you're advocating changing `ALPHANUMERIC_EXPANSION_PATTERN` from the current `r'\[((?:[a-zA-Z0-9]+[?:,-])+[a-zA-Z0-9]+)\]'` to something like `r'\[[^]*\]'`, which basically matches anything in brackets, even if it's never going to be parsable.

This would be possible as a simplification, but it would be a potentially breaking change, since it'll change how ranges are parsed, so I think it's best to leave this one alone.